src/redux/setupListenersReactNative.js

```diff
@@ -0,0 +1,60 @@
+import NetInfo from '@react-native-community/netinfo';
+import { AppState } from 'react-native';
+
+let initialized = false;
+
+let appStateSubscription = null;
+let appState = AppState.currentState;
+
+let netInfoUnsubscribe = null;
+
+export function setupListenersReactNative(
+  dispatch,
+  { onFocus, onFocusLost, onOffline, onOnline },
+) {
+  const handleFocus = () => dispatch(onFocus());
+  const handleFocusLost = () => dispatch(onFocusLost());
+  const handleOnline = () => dispatch(onOnline());
+  const handleOffline = () => dispatch(onOffline());
+
+  const _handleAppStateChange = nextAppState => {
+    const foreground = !!(
```

Comment:
double `!` is negating the negation thus "useless"?

Reply:
no, it's a fancy way to convert the value to boolean